docs/collection_metadata.rst

```diff
@@ -3,30 +3,77 @@
 Collection-level dependencies
 =============================
 
+.. note::
+
+  The Ansible Builder's dependency introspection described in this document is not a part of a typical workflow
+  for building and using execution environments. It is a troubleshooting functionality for collection maintainers
```

Comment:
Not entirely accurate. If any requirements are given (galaxy, python, or system), introspection is run so that the requirements are sanitized (deduped, basically) before the `assemble` steps. A user may see the output from this during the builder intermediate phase when using the `build -v3` option. This is also a good way to discover where conflicting Python requirements are coming from since `introspect` will output that information.

Reply:
@Shrews thanks for the clarification, much appreciated
How about now? the last two commits